core/tests/test_utils.py

```diff
@@ -1251,22 +1251,23 @@ def get_user_id_from_email(self, email):
             email: str. A valid email stored in the App Engine database.
 
         Returns:
-            str. ID of the user possessing the given email.
+            str or None. ID of the user possessing the given email, or None if
+            the user does not exist.
         """
-        gae_id = self.get_gae_id_from_email(email)
-        return (
-            user_services.get_user_settings_by_gae_id(gae_id).user_id)
+        user_settings = user_services.get_user_settings_by_gae_id(
+            self.get_gae_id_from_email(email))
+        return user_settings and user_settings.user_id
 
     def get_gae_id_from_email(self, email):
-        """Gets the GAE user ID corresponding to the given email.
+        """Mocks a GAE user ID corresponding to the given email.
 
         Args:
-            email: str. A valid email stored in the App Engine database.
+            email: str. The email address of the user.
 
         Returns:
-            str. GAE ID of the user possessing the given email.
+            str. Mocked GAE ID for the user possessing the given email.
         """
-        return current_user_services.get_gae_id_from_email(email)
+        return python_utils.convert_to_bytes(hash(email))
```

Comment:
Could you either (a) add some explanation that this function can be arbitrary as long as different emails are mapped to different mocked GAE IDs, or (b) point to a reference explaining why this method of ID generation is correct in the testing context?

Reply:
Done.